samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -76,108 +76,182 @@ private enum SupportedFileType
+                documentMessageContent,
+                documentImportForm);
+            if (chatMessage == null)
+            {
+                // It's Ok to have the message not created.
+                return this.Ok();
+            }
+
+            var chatId = documentImportForm.ChatId.ToString();
+            await messageRelayHubContext.Clients.Group(chatId)
+                .SendAsync(ChatDocumentUploadedClientCall, chatMessage, chatId);
+
+            return this.Ok(chatMessage);
         }
 
+        await messageRelayHubContext.Clients.All.SendAsync(
+            GlobalDocumentUploadedClientCall,
+            documentMessageContent.ToFormattedStringNamesOnly(),
+            documentImportForm.UserName
+        );
+
+        return this.Ok();
+    }
+
+    #region Private
+
+    /// <summary>
+    /// Validates the document import form.
+    /// </summary>
+    /// <param name="documentImportForm">The document import form.</param>
+    /// <returns></returns>
+    /// <exception cref="ArgumentException">Throws ArgumentException if validation fails.</exception>
+    private async Task ValidateDocumentImportFormAsync(DocumentImportForm documentImportForm)
+    {
+        var formFiles = documentImportForm.FormFiles;
+
+        if (formFiles.Count == 0)
+        {
+            throw new ArgumentException("No file was uploaded.");
+        }
+        else if (formFiles.Count > this._options.FileCountLimit)
+        {
+            throw new ArgumentException($"Too many files uploaded. Max file count is {this._options.FileCountLimit}.");
+        }
+
+        // Loop through the uploaded files and validate them before importing.
+        foreach (var formFile in formFiles)
+        {
+            if (formFile.Length == 0)
+            {
+                throw new ArgumentException($"File {formFile.FileName} is empty.");
+            }
+
+            if (formFile.Length > this._options.FileSizeLimit)
+            {
+                throw new ArgumentException($"File {formFile.FileName} size exceeds the limit.");
+            }
+
+            // Make sure the file type is supported.
+            var fileType = this.GetFileType(Path.GetFileName(formFile.FileName));
+            switch (fileType)
+            {
+                case SupportedFileType.Txt:
+                case SupportedFileType.Pdf:
+                    break;
+                default:
+                    throw new ArgumentException($"Unsupported file type: {fileType}");
+            }
+        }
+
+        // Make sure the user has access to the chat session if the document is uploaded to a chat session.
         if (documentImportForm.DocumentScope == DocumentImportForm.DocumentScopes.Chat
-            && !(await this.UserHasAccessToChatAsync(documentImportForm.UserId, documentImportForm.ChatId)))
+                && !(await this.UserHasAccessToChatAsync(documentImportForm.UserId, documentImportForm.ChatId)))
```

Comment:
nit: Move this check earlier in the function. Better to early out on this than after validating 10 files